appium/lib/screen-object.rb

```diff
@@ -69,81 +66,80 @@ def enter
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
+  def scroll(direction = :down, duration = 1000)
+    size = driver.window_size
+    x = size.width/2
+    y = size.height/2
+    case direction
+    when :up then loc = [x, y*0.5, x, (y + (y*0.5)),duration]
+    when :down then loc =[x, y, x, y * 0.5,duration]
+    when :left then loc = [x * 0.6, y, x * 0.3, y,duration]
+    when :right then loc =[x * 0.3, y, x * 0.6, y,duration]
+    else
+      raise('Only upwards and downwards scrolling are supported')
     end
+    gesture(loc[0],loc[1],loc[2],loc[3],loc[4])
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def scroll_down
+    scroll(:down)
+  end
+
+  def scroll_up
+    scroll(:up)
+  end
+
+  def swipe_left
+    scroll(:left)
   end
 
+  def swipe_right
+    scroll(:right)
+  end
 
-  def scroll_up_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
+  def swipe_element(locator, direction = :down, duration = 1000)
+    my_element = driver.find_element(locator.locator.first,locator.locator.last).rect
+    start_x = my_element.x
+    end_x = my_element.x + my_element.width
+    start_y = my_element.y
+    end_y = my_element.y + my_element.height
+    height = my_element.height
+    case direction
+    when :up then  loc = [end_x * 0.5, (start_y + ( height * 0.2)), end_x * 0.5, (end_y - (height * 0.2)),duration]
+    when :down then  loc = [end_x * 0.5, (end_y - (height * 0.2)), start_x * 0.5, (start_y + (height * 0.3)), duration]
+    when :left then  loc = [end_x * 0.9,  end_y - (height/2), start_x, end_y - (height/2), duration]
+    when :right then  loc = [end_x * 0.1, end_y - (height/2), end_x * 0.9, end_y - (height/2), duration]
+    else
+      raise('Only upwards and downwards scrolling are supported')
+    end
+    gesture(loc[0],loc[1],loc[2],loc[3],loc[4])
+  end
+  
+  def scroll_find(locator, direction = 'down', num_loop = 15)
+    driver.manage.timeouts.implicit_wait = 1
```

Comment:
If you are adjusting the `implicit_wait` for this method, make sure when the method is complete, you are setting it back to what it was initially. Capture the value of `driver.manage.timeouts.implicit_wait` and reset it before this method exits.